pkg/interactor/data.go

```diff
@@ -52,8 +53,27 @@ func (d *DataInteractor) Save(data Data) error {
 	err := d.DataStore.Save(data)
 	if err != nil {
 		d.logger.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (d *DataInteractor) DeleteByDeviceID(deviceId string) error {
```

Comment:
Notice your changes here are related to the interactor layer instead of the database. So, please adjust the commit message to describe this.

Reply:
Done!